crates/smdk/src/generate.rs

```diff
@@ -0,0 +1,119 @@
+use anyhow::{Error, Result};
+use clap::Parser;
+use cargo_generate::{GenerateArgs, TemplatePath, generate};
+use include_dir::{Dir, include_dir};
+use tempdir::TempDir;
+
+static SMART_MODULE_TEMPLATE: Dir<'static> =
+    include_dir!("$CARGO_MANIFEST_DIR/../../smartmodule/cargo_template");
+
+/// Generate new SmartModule project
+#[derive(Debug, Parser)]
+pub struct GenerateOpt {
+    /// SmartModule Project Name
+    name: String,
+    /// Template to generate project from.
+    ///
+    /// Must be a GIT repository
+    #[clap(long)]
+    template: Option<String>,
+}
+
+/// Abstraction on different of template options available for generating a
+/// new SmartModule project.
+///
+/// May hold a reference to a `TempDir` which should not be dropped before
+/// accomplishing the project generation procedure.
+struct Template {
+    template_path: TemplatePath,
+    _temp_dir: Option<TempDir>,
+}
+
+impl Template {
+    /// Extracts inlined directory contents into a temporary directory and
+    /// builds a `TemplatePath` instance with the `path` pointing to the temp
+    /// directory created.
+    ///
+    /// Is important to hold the reference to the `_temp_dir` until generation
+    /// process is completed, otherwise the temp directory will be deleted
+    /// before reaching the generation process.
+    fn inline() -> Result<Self> {
```

Comment:
let's add test to ensure inline unpacking works

Reply:
Of course, here [d6987fd](https://github.com/infinyon/fluvio/pull/2630/commits/d6987fdd490df93f63c8c770749c05b9f10d5b15) we have tests for each step of the process, making sure the temp directory is created and also one of the files from the template `Smart.toml` is available in such temp directory as well.